src/fastga_he/models/propulsion/components/loads/sm_pmsm/components/perf_iron_losses.py

```diff
@@ -0,0 +1,167 @@
+            "number_of_points", default=1, desc="number of equilibrium to be treated"
+        )
+
+    def setup(self):
+        motor_id = self.options["motor_id"]
+        number_of_points = self.options["number_of_points"]
+
+        self.add_input(
+            "electrical_frequency",
+            units="Hz",
+            val=np.nan,
+            shape=number_of_points,
+            desc="The oscillation frequency of the SM PMSM AC current",
+        )
+        self.add_input(
+            name="data:propulsion:he_power_train:SM_PMSM:"
+            + motor_id
+            + ":design_air_gap_flux_density",
+            val=np.nan,
+            units="T",
+            desc="The design air gap magnetic flux density",
+        )
+        self.add_input(
+            name="data:propulsion:he_power_train:SM_PMSM:" + motor_id + ":mass",
+            val=np.nan,
+            units="kg",
+        )
+
+        self.add_output(
+            "iron_power_losses",
+            units="kW",
+            val=0.0,
+            shape=number_of_points,
+            desc="Iron losses of the SM PMSM due to altering magnetic flux",
+        )
+
+    def setup_partials(self):
+        motor_id = self.options["motor_id"]
+        number_of_points = self.options["number_of_points"]
+
+        self.declare_partials(
+            of="iron_power_losses",
+            wrt=["electrical_frequency"],
+            method="exact",
+            rows=np.arange(number_of_points),
+            cols=np.arange(number_of_points),
+        )
+        self.declare_partials(
+            of="iron_power_losses",
+            wrt=[
+                "data:propulsion:he_power_train:SM_PMSM:"
+                + motor_id
+                + ":design_air_gap_flux_density",
+                "data:propulsion:he_power_train:SM_PMSM:" + motor_id + ":mass",
+            ],
+            method="exact",
+            rows=np.arange(number_of_points),
+            cols=np.zeros(number_of_points),
+        )
+
+    def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+        motor_id = self.options["motor_id"]
+
+        mass = inputs["data:propulsion:he_power_train:SM_PMSM:" + motor_id + ":mass"]
+        air_gap_flux_density = inputs[
+            "data:propulsion:he_power_train:SM_PMSM:" + motor_id + ":design_air_gap_flux_density"
+        ]
+        f = inputs["electrical_frequency"]
+
+        # Pre-calculate common terms
+        f_powers = np.sqrt(f) ** np.arange(1, 5)[:, np.newaxis]
+        air_gap_flux_density_powers = np.sqrt(air_gap_flux_density) ** np.arange(1, 5)
+
+        # Broadcasting automatically handles the multiplication
+        outputs["iron_power_losses"] = (
+            mass
+            * np.sum(
+                np.array(IRON_LOSSES_COEFF)[:, :, np.newaxis]
+                * f_powers[:, np.newaxis, :]
+                * air_gap_flux_density_powers[np.newaxis, :, np.newaxis],
+                axis=(0, 1),
+            )
```

Comment:
Maybe it is just me, but I find that quite unreadable ^^' It is a polynomial no ? I understand it is more compact like this but wouldn't it be better explicited ? It would spare us an import of iron losses coeff as well